### ISSUE TITLE
Syntax to refer explicitly to the current module

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-spec#96]
It would occasionally be useful to have a syntax to refer explicitly to the current module. For example, in a context where a local variable x is shadowing a module level variable x, the latter could be referred to by XXX:x, where XXX is the syntax for referring explicitly to the current module.

This might also help with annotations, where we want an unqualified annotation tag to refer to a built-in annotation tag rather than a tag defined in the the current module.

Alternatively we could provide a way for import to declare a module name as referring to the current module.

**Comment by sanjiva** [comment]
Yes I was thinking about this too .. maybe reuse `self`? Something like:

`import self as x;`

We can't use `self` directly because of its use within objects for a different purpose.

**Comment by jclark** [comment]
If module aliases are a different symbol space from the main symbol space, then we could potentially reuse `self` directly. See #97.

**Comment by sameerajayasoma** [comment]
In `self.foo`, self refers to the current object instance and in `self:bar`, self refers to the current module *instance*. I think it is okay to allow `self` directly given that module aliases and XML namespace names are in a separate symbol space. 

In fact, at runtime, a Ballerina module can be considered as a value which has a state and a set of associated functions. :)
